src/tiles.js

```diff
@@ -0,0 +1,152 @@
+import keys from 'lodash/keys';
+import includes from 'lodash/includes';
+import isUndefined from 'lodash/isUndefined';
+import { assert } from './utils/asserts';
+import { amBlue, amRed } from './helpers/player';
+
+let tileInfo;
+let tileNames;
+
+/*
+ * Stores all information we need about tiles in the tileInfo object, and creates the tileNames
+ * object so that we can quickly map from an id to a name. This function depends on amBlue and
+ * amRed, so the current player must be defined. We call this function once, after our player has an
+ * id.
+ */
+export function computeTileInfo() {
```

Comment:
There's a lag between when the client side javascript loads, and when our player has an id. We only compute all the info about the tiles once. If we just assigned it to the dictionary that's in this function, then we would store all the info about tiles immediately when the client side JS loads, and it wouldn't know if a `RED_GATE` is traversable, for example. The solution is to put the initialization of `tileInfo` in a function, and then call that function once our player has an `id`. Look at line 43 in `index.js`, thats when we initialize `tileInfo`